### PR TITLE
EFK for 19.10

### PR DESCRIPTION
**Pull request by fatpat**

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -6,31 +6,57 @@
     - "{{ ansible_os_family }}.yml"
   tags: install
 
-- name: Generate configuration file
+- name: Test directories presence
+  stat:
+    path: "{{ item }}"
+  register: _dir
+  with_items:
+    - "/etc/gridinit.d/{{ openio_kibana_namespace }}"
```
Comment (cdelgehier) [inline tasks/main.yml]: use
```
    - name: Ensure namespace directories exist
      file:
        path: "{{ item }}"
        state: directory
        owner: openio
        group: openio
        mode: "0755"
      with_items:
        - "/etc/gridinit.d/{{ openio_kibana_namespace }}"
        …
```

File: tasks/main.yml [hunk 2]
```diff
@@ -6,31 +6,57 @@
...
     - _kibana_conf is changed
+    - not openio_kibana_provision_only
 
-- name: Ensure kibana is listening
+- name: Wait for kibana to be available
   wait_for:
     host: "{{ openio_kibana_bind_address }}"
     port: "{{ openio_kibana_bind_port }}"
     delay: 10
-  tags: configure
```
Comment (cdelgehier) [inline tasks/main.yml:33]: tasks must be tagged install or configure
rpm and mkdir are `install`
anything else is configure

File: vars/main.yml [hunk 3]
```diff
@@ -1,3 +1,4 @@
 ---
-openio_kibana_version: "6.x"
+openio_kibana_version: "7.x"
+openio_kibana_servicename: "kibana-{{ openio_kibana_serviceid }}"
```
Comment (cdelgehier) [inline vars/main.yml:3]: define it in vars/main.yml

**Review by cdelgehier — CHANGES_REQUESTED** [review]

File: defaults/main.yml [hunk 4]
```diff
@@ -1,12 +1,23 @@
 ---
+openio_kibana_namespace: "OPENIO"
```
Comment (cdelgehier) [inline defaults/main.yml]: openio_kibana_namespace: "{{ namespace | default('OPENIO') }}"

File: defaults/main.yml [hunk 5]
```diff
@@ -1,12 +1,23 @@
...
+openio_kibana_serviceid: "0"
+
+openio_kibana_gridinit_dir: "/etc/gridinit.d"
+openio_kibana_gridinit_file_prefix: "{{ openio_kibana_namespace }}-"
+
+openio_kibana_sysconfig_dir: "/etc/oio/sds/{{ openio_kibana_namespace }}/{{ openio_kibana_servicename }}"
```
Comment (cdelgehier) [inline defaults/main.yml]: all of 3 are vars, not defaults

File: defaults/main.yml [hunk 6]
```diff
@@ -1,12 +1,23 @@
...
+openio_kibana_log_dir: "/var/log/oio/sds/{{ openio_kibana_namespace }}/{{ openio_kibana_servicename }}"
+openio_kibana_pid_dir: "/run/oio/sds/{{ openio_kibana_namespace }}"
```
Comment (cdelgehier) [inline defaults/main.yml]: `openio_kibana_pid_directory` like in other roles

File: defaults/main.yml [hunk 7]
```diff
@@ -1,12 +1,23 @@
...
+
+openio_kibana_volume:
+  "/var/lib/oio/sds/{{ openio_kibana_namespace }}/kibana-{{ openio_kibana_serviceid }}"
+
 openio_kibana_provision_only: false
 openio_kibana_service_enabled: true
 openio_kibana_bind_port: 5601
```
Comment (cdelgehier) [inline defaults/main.yml]: 6000 < openio_kibana_bind_port   < 6999

File: defaults/main.yml [hunk 8]
```diff
@@ -1,12 +1,23 @@
...
-openio_kibana_confdir: "/etc/kibana"
 openio_kibana_bind_interface: "{{ openio_bind_mgmt_interface | d(ansible_default_ipv4.alias) }}"
-openio_kibana_bind_address: "{{ openio_bind_mgmt_address \
-      | default(hostvars[inventory_hostname]['ansible_' + openio_kibana_bind_interface]['ipv4']['address']) }}"
-openio_kibana_ext_bind_address: ""
+openio_kibana_bind_address: "{{ openio_bind_mgmt_address
+  | d(hostvars[inventory_hostname]['ansible_' + openio_kibana_bind_interface].ipv4.address) }}"
```
Comment (cdelgehier) [inline defaults/main.yml]: | default(hostvars[inventory_hostname]['ansible_' + openio_kibana_bind_interface]['ipv4']['address']

Don't mix format
Choose the dot style or array style, not both

File: tasks/main.yml [hunk 9]
```diff
@@ -6,36 +6,73 @@
...
+
+- name: Ensure Kibana pid directory is persistant
+  lineinfile:
+    dest: /etc/tmpfiles.d/kibana.conf
+    create: true
+    line: "d {{ openio_kibana_pid_dir }} 755 openio openio -"
+    owner: openio
+    group: openio
+    mode: 0644
+  tags: install
```
Comment (cdelgehier) [inline tasks/main.yml]: no, configure

File: tasks/main.yml [hunk 10]
```diff
@@ -6,36 +6,73 @@
...
     owner: openio
     group: openio
     mode: 0644
+  with_items:
+    - src: "kibana.yml.j2"
+      dest: "{{ openio_kibana_sysconfig_dir }}/kibana.yml"
+    - src: "gridinit_kibana.conf.j2"
+      dest: "{{ openio_kibana_gridinit_dir }}/{{ openio_kibana_gridinit_file_prefix}}{{ openio_kibana_servicename }}.conf"
   register: _kibana_conf
-  tags: configure
```
Comment (cdelgehier) [inline tasks/main.yml:17]: need a tag

File: tasks/main.yml [hunk 11]
```diff
@@ -6,36 +6,73 @@
...
     host: "{{ openio_kibana_bind_address }}"
     port: "{{ openio_kibana_bind_port }}"
     delay: 10
-  tags: configure
 
 - name: Create index pattern
   uri:
-    url: "http://{{ openio_kibana_bind_address }}:{{ openio_kibana_bind_port }}\
-/api/saved_objects/index-pattern/oio-filebeat"
+    url: "{{ openio_kibana_url }}/api/saved_objects/index-pattern/oio-filebeat"
```
Comment (cdelgehier) [inline tasks/main.yml:84]: need retry

File: tasks/main.yml [hunk 12]
```diff
@@ -54,7 +91,7 @@
 
 - name: Set index pattern as default
   uri:
-    url: "http://{{ openio_kibana_bind_address }}:{{ openio_kibana_bind_port }}/api/kibana/settings/defaultIndex"
+    url: "{{ openio_kibana_url }}/api/kibana/settings/defaultIndex"
```
Comment (cdelgehier) [inline tasks/main.yml:104]: need retry

**Review by cdelgehier — CHANGES_REQUESTED** [review]

File: tasks/main.yml (hunk 1 above)
Comment (fatpat) [reply tasks/main.yml]: otudated

File: vars/main.yml (hunk 3 above)
Comment (fatpat) [reply vars/main.yml:3]: ? it is already in `vars/main.yml`

File: defaults/main.yml (hunk 4 above)
Comment (fatpat) [reply defaults/main.yml]: fixed in the 3 roles

File: defaults/main.yml (hunk 6 above)
Comment (fatpat) [reply defaults/main.yml]: so `sysconfig`, `log` are suffixed with `_dir` but `pid` is suffixed with `_directory` 

where is the consistency here ?

i'll fix this for those 3 EFK roles but this does not make sens globally IMHO

Comment (fatpat) [reply defaults/main.yml]: fixed

File: tasks/main.yml (hunk 2 above)
Comment (fatpat) [reply tasks/main.yml:33]: fixed

File: vars/main.yml (hunk 3 above)
Comment (fatpat) [reply vars/main.yml:3]: fixed

File: defaults/main.yml (hunk 5 above)
Comment (fatpat) [reply defaults/main.yml]: fixed

File: defaults/main.yml (hunk 7 above)
Comment (fatpat) [reply defaults/main.yml]: fixed

File: defaults/main.yml (hunk 8 above)
Comment (fatpat) [reply defaults/main.yml]: fixed

File: tasks/main.yml (hunk 9 above)
Comment (fatpat) [reply tasks/main.yml]: fixed

File: tasks/main.yml (hunk 10 above)
Comment (fatpat) [reply tasks/main.yml:17]: fixed

File: tasks/main.yml (hunk 11 above)
Comment (fatpat) [reply tasks/main.yml:84]: fixed

File: tasks/main.yml (hunk 12 above)
Comment (fatpat) [reply tasks/main.yml:104]: fixed

**Pull request merged**